infra/Pos/Slotting/Impl/Simple.hs

```diff
@@ -32,7 +34,14 @@ type SimpleSlottingMode ctx m
     = ( Mockable CurrentTime m
       , MonadSlotsData ctx m
       , MonadIO m
-      , HasConfiguration)
+      , HasConfiguration
```

Comment:
This seems dubious. I wonder what it actually needs... probably only `ProtocolConstants`, because slotting stuff depends upon it. OK to leave it for now, but after https://github.com/input-output-hk/cardano-sl/pull/2590 is merged it can probably be loosened to `HasProtocolConstants`.